x-pack/plugins/case/common/api/cases/comment.ts

```diff
@@ -18,22 +33,42 @@ export const CommentAttributesBasicRt = rt.type({
   updated_by: rt.union([UserRT, rt.null]),
 });
 
+export enum CommentType {
+  user = 'user',
+  alert = 'alert',
+  generatedAlert = 'generated_alert',
+}
+
 export const ContextTypeUserRt = rt.type({
   comment: rt.string,
-  type: rt.literal('user'),
+  type: rt.literal(CommentType.user),
 });
 
 export const ContextTypeAlertRt = rt.type({
-  type: rt.literal('alert'),
-  alertId: rt.string,
+  type: rt.union([rt.literal(CommentType.generatedAlert), rt.literal(CommentType.alert)]),
+  alertId: rt.union([rt.array(rt.string), rt.string]),
+  index: rt.string,
+});
+
+const AlertIDRt = rt.type({
+  _id: rt.string,
+});
+
+export const ContextTypeGeneratedAlertRt = rt.type({
+  type: rt.literal(CommentType.generatedAlert),
+  alerts: rt.union([rt.array(AlertIDRt), AlertIDRt]),
```

Comment:
Why we need to have both an array of strings and a string as type?

Reply:
Good point. This one could probably be switched to just `AlertIDRt` since this is the first time we're introducing this type but I figured we'd just keep it similar to the regular alerts and support both. I chatted with Xavier and for now I think we're going to keep it as `string | string[]` because we'd have to do a migration on the old saved objects if they have and alert to make them a `string[]`. We could do that but we're kind of short on time 😬 . I'll add it to the list to refactor 👍